axon-server-connector/src/main/java/org/axonframework/axonserver/connector/query/AxonServerQueryBus.java

```diff
@@ -249,8 +250,28 @@ public <R> Registration subscribe(@Nonnull String queryName,
         return new AxonServerRegistration(localRegistration, serverRegistration::cancel);
     }
 
+    private <Q, R> CompletableFuture<QueryResponseMessage<R>> tryToRunQueryLocally(
+            @Nonnull QueryMessage<Q, R> queryMessage) {
+
+        if (localSegment instanceof SimpleQueryBus) {
```

Comment:
Instead, we should have a configurable boolean check here, as it should be a choice for the user of the `AxonServerQueryBus` to short-cut to the `localSegment`.

The reasoning for that is that we cannot assume that the shortcut is what every user needs.

Introducing that boolean would start by adding it to the `AxonServerQueryBus.Builder`. For example, by adding the `AxonServerQueryBus.Builder#localSegmentShortCut()` merthod. This method would then set a `boolean` in the `AxonServerQueryBus.Builder`, which the `AxonServerQueryBus` can set as global state for the bus.